variables.tf

```
@@ -0,0 +1,140 @@
+    kerberosArmoring      = optional(bool, true)
+    kerberosRc4Encryption = optional(bool, false)
+    ldapSigning           = optional(bool, true)
+    ntlmV1                = optional(bool, false)
+    syncKerberosPasswords = optional(bool, true)
+    syncNtlmPasswords     = optional(bool, true)
+    syncOnPremPasswords   = optional(bool, true)
+    tlsV1                 = optional(bool, false)
+  })
+  default     = {}
+  description = <<-DOC
+  ```
+    channelBinding: Require all LDAP clients to provide channel binding information when communicating with the directory. Any client that does not provide this information will fail.
+    kerberosArmoring: Enable or disable Kerberos Armoring for your managed domain. This will provide a protected channel between the Kerberos client and the KDC.
+    kerberosRc4Encryption: Enable or disable Kerberos RC4 encryption for your managed domain. When Kerberos RC4 encryption is disabled, all Kerberos requests that use RC4 encryption will fail.
+    ldapSigning: Require all LDAP clients to request signing during bind time. Any bind request that does not request signing will fail.
+    ntlmV1: Enable or disable NTLM v1 authentication for your managed domain. When NTLM v1 authentication is disabled, all NTLM v1 authentication requests will fail.
+    syncKerberosPasswords: Enable or disable Kerberos password hash synchronization. When this is disabled, synchronized user accounts will be unable to use Kerberos authentication in the managed domain.
+    syncNtlmPasswords: Enable or disable NTLM password hash synchronization. When this is disabled, synchronized user accounts will be unable to use NTLM authentication in the managed domain.
+    syncOnPremPasswords: Enable or disable password hash synchronization for on-premises accounts. When this is disabled, on-premises user accounts will be unable to authenticate in the managed domain. Cloud-only users will be unaffected.
+    tlsV1: Enable or disable TLS 1 legacy mode for your managed domain.
+  ```
+  DOC
+}
+
+variable "filtered_sync" {
+  type        = bool
+  default     = true
+  description = "Enabled or Disabled flag to turn on Group-based filtered sync"
+}
+
+variable "ldaps_settings" {
+  type = object({
+    externalAccess         = optional(bool, false)
+    ldaps                  = bool
+    pfxCertificate         = optional(string, "")
+    pfxCertificatePassword = optional(string, "")
+  })
+  description = <<-DOC
+  ```
+    externalAccess: A flag to determine whether or not Secure LDAP access over the internet is enabled or disabled.	
+    ldaps: A flag to determine whether or not Secure LDAP is enabled or disabled.
+    pfxCertificate: The certificate required to configure Secure LDAP. The parameter passed here should be a base64encoded representation of the certificate pfx file.
+    pfxCertificatePassword: The password to decrypt the provided Secure LDAP certificate pfx file.
+  ```
+  DOC
+}
```

````suggestion
variable "ldaps_settings" {
  type = object({
    externalAccess         = optional(bool, false)
    pfxCertificate         = string
    pfxCertificatePassword = string
  })

  sensitive = true

  description = <<-DOC
  Configure LDAPS. To disable LDAPS, set the configuration to `null`.
  ```
    externalAccess: A flag to determine whether or not Secure LDAP access over the internet is enabled or disabled.	
    pfxCertificate: Base64encoded representation certificate required to configure Secure LDAP.
    pfxCertificatePassword: The password to decrypt the provided Secure LDAP certificate pfx file.
  ```
  DOC
}
````

We should be secure by default, so requiring LDAPS.